openbas-framework/src/main/java/io/openbas/execution/Injector.java

```diff
@@ -30,227 +30,232 @@
-        expectationExecution.setInject(executableInject.getInjection().getInject());
-        expectationExecution.setExpectedScore(expectation.getScore());
-        expectationExecution.setExpectationGroup(expectation.isExpectationGroup());
-        switch (expectation.type()) {
-            case ARTICLE -> {
-                expectationExecution.setName(expectation.getName());
-                expectationExecution.setArticle(((ChannelExpectation) expectation).getArticle());
-            }
-            case CHALLENGE -> {
-                expectationExecution.setName(expectation.getName());
-                expectationExecution.setChallenge(((ChallengeExpectation) expectation).getChallenge());
-            }
-            case DOCUMENT -> expectationExecution.setType(EXPECTATION_TYPE.DOCUMENT);
-            case TEXT -> expectationExecution.setType(EXPECTATION_TYPE.TEXT);
-            case DETECTION -> {
-                DetectionExpectation detectionExpectation = (DetectionExpectation) expectation;
-                expectationExecution.setName(detectionExpectation.getName());
-                expectationExecution.setDetection(detectionExpectation.getAsset(), detectionExpectation.getAssetGroup());
-                expectationExecution.setSignatures(detectionExpectation.getInjectExpectationSignatures());
-            }
-            case PREVENTION -> {
-                PreventionExpectation preventionExpectation = (PreventionExpectation) expectation;
-                expectationExecution.setName(preventionExpectation.getName());
-                expectationExecution.setPrevention(preventionExpectation.getAsset(), preventionExpectation.getAssetGroup());
-                expectationExecution.setSignatures(preventionExpectation.getInjectExpectationSignatures());
-            }
-            case MANUAL -> {
-                ManualExpectation manualExpectation = (ManualExpectation) expectation;
-                expectationExecution.setName(((ManualExpectation) expectation).getName());
-                expectationExecution.setManual(manualExpectation.getAsset(), manualExpectation.getAssetGroup());
-                expectationExecution.setDescription(((ManualExpectation) expectation).getDescription());
-            }
-            default -> throw new IllegalStateException("Unexpected value: " + expectation);
-        }
-        return expectationExecution;
+  private InjectExpectation expectationConverter(
+      @NotNull InjectExpectation expectationExecution,
+      @NotNull final ExecutableInject executableInject,
+      @NotNull final Expectation expectation) {
+    expectationExecution.setExercise(executableInject.getInjection().getExercise());
+    expectationExecution.setInject(executableInject.getInjection().getInject());
+    expectationExecution.setExpectedScore(expectation.getScore());
+    expectationExecution.setExpectationGroup(expectation.isExpectationGroup());
+    expectationExecution.setExpirationTime(expectation.getExpirationTime());
```

Comment:
We need to add a default value here like 0 to avoid DB error.
When an inject already exists, the expectations are in the content and are not handle by the migration.

expectationExecution.setExpirationTime(Optional.ofNullable(expectation.getExpirationTime()).orElse(0L));